api/undertaker/undertaker.go

```diff
@@ -0,0 +1,132 @@
+package undertaker
+
+import (
+	"time"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/api"
+	"github.com/juju/juju/api/base"
+	"github.com/juju/juju/api/common"
+	"github.com/juju/juju/apiserver/params"
+)
+
+// Client provides access to the undertaker api
+type Client struct {
+	base.ClientFacade
+	st     api.Connection
+	facade base.FacadeCaller
+}
+
+// UnderTakerClient defines the methods on the undertaker API end point.
+type UndertakerClient interface {
+	ProcessDyingEnviron() error
+	RemoveAllEnvironDocs() error
```

Comment:
I don't think an external API should refer to internal concerns such as documents. Alternatives:
* RemoveEnvironData
* DeleteEnvironData
* RemoveEnviron 
* DeleteEnviron
* FinaliseEnviron
* Finalise